taskvine/src/manager/vine_manager.c

```diff
@@ -2602,154 +2603,117 @@ struct rmsummary *vine_manager_choose_resources_for_task(struct vine_manager *q,
 		return limits;
 	}
 
-	if (t->input_files_size < 0) {
-		vine_manager_compute_input_size(q, t);
+	/* Shortcut if the task explicitly requests all resources */
```

Comment:
This is a large rework of a critical function, can you explain in more detail what this accomplishes?